### PR TITLE
Addresses #1025: Added some tests for Infobox Component

### PR DESCRIPTION
**Pull request by praveenojha33**
Addresses #1025
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)
- [x] Added Surge preview link

#### Changes proposed in this pull request:
Added some tests for Infobox Component.
Surge:https://pr-1032-fossasia-susper.surge.sh

### CODE REVIEW

**Comment by praveenojha33** [comment]
@simsausaurabh Please review.

**Comment by praveenojha33** [comment]
Coverage is decreasing currently. I will be updating this PR to improve tests.

**Comment by praveenojha33** [comment]
@simsausaurabh @ValentinBercaru Please review.

File: src/app/infobox/infobox.component.spec.ts [hunk 1]
```diff
@@ -1,65 +1,57 @@
...
-        FooterNavbarComponent,
-        AboutComponent,
-        ContactComponent,
-        ModalComponent,
-        InfoboxComponent ],
+        InfoboxComponent,
+      ],
       providers: [
-        KnowledgeapiService
+        { provide: KnowledgeapiService, useValue: MockKnowledgeApi},
```
Comment (simsausaurabh) [inline src/app/infobox/infobox.component.spec.ts]: Keep proper spacing between curly braces and imported elements. Please follow styles.

File: src/app/infobox/infobox.component.spec.ts [hunk 2]
```diff
@@ -1,65 +1,57 @@
...
     TestBed.configureTestingModule({
       imports: [
         RouterTestingModule,
-        BrowserModule,
-        CommonModule,
-        FormsModule,
         HttpModule,
         JsonpModule,
         StoreModule.provideStore(reducer),
-        StoreDevtoolsModule.instrumentOnlyWithExtension(),
```
Comment (simsausaurabh) [inline src/app/infobox/infobox.component.spec.ts]: Why did you remove StoreDevtoolsModule import?

File: src/app/infobox/infobox.component.spec.ts [hunk 3]
```diff
@@ -1,65 +1,57 @@
...
-import {KnowledgeapiService} from "../services/knowledgeapi.service";
-import {reducer} from "../reducers/index";
-import {StoreModule} from "@ngrx/store";
-import {StoreDevtoolsModule} from "@ngrx/store-devtools";
+import { RouterTestingModule } from "@angular/router/testing";
+import { ThemeService } from '../services/theme.service';
+import { KnowledgeapiService } from "../services/knowledgeapi.service";
+import { reducer } from "../reducers/index";
+import { StoreModule } from "@ngrx/store";
+import {MockKnowledgeApi} from "../shared/mocks/knowledge.mock";
```
Comment (simsausaurabh) [inline src/app/infobox/infobox.component.spec.ts]: Keep proper style which importing modules. Keep spacing properly.

Comment (simsausaurabh) [inline src/app/infobox/infobox.component.spec.ts]: Removing other imports should not make it as addition of new modules. Just remove all other unused imports.

File: src/app/infobox/infobox.component.ts [hunk 4]
```diff
@@ -24,11 +25,9 @@ export class InfoboxComponent implements OnInit {
     this.content_response$ = store.select(fromRoot.getKnowledgeContent);
     this.content_response$.subscribe(res => {
       if (res.extract) {
+        this.results = res;
```
Comment (simsausaurabh) [inline src/app/infobox/infobox.component.ts]: results should be optional? (```results$``` ?)

File: src/app/infobox/infobox.component.spec.ts (hunk 2 above)
Comment (praveenojha33) [reply src/app/infobox/infobox.component.spec.ts]: It was not needed.

File: src/app/infobox/infobox.component.spec.ts (hunk 3 above)
Comment (praveenojha33) [reply src/app/infobox/infobox.component.spec.ts]: Some imports which are not needed are used by `Components` during testing. Removing them will show error while testing.

Comment (simsausaurabh) [reply src/app/infobox/infobox.component.spec.ts]: I meant to say - You are actually removing the imports which are not needed right? Then it should only tell that you are actually removing and not adding anything (Here it seems you are adding these lines. Try to remove the lines one by one and it shall not show.)

File: src/app/infobox/infobox.component.spec.ts (hunk 2 above)
Comment (simsausaurabh) [reply src/app/infobox/infobox.component.spec.ts]: It is used to check test the dev tools with the Chrome Extension. To check if its not needed, do one thing - Enable StoreDevtools and run the tests alongside the project. See if it passes. If it passes, then it is okay to remove it.

File: src/app/infobox/infobox.component.spec.ts (hunk 3 above)
Comment (praveenojha33) [reply src/app/infobox/infobox.component.spec.ts]: Removing the imports is causing tests to fail as these imports will be used by Infobox components while testing.

File: src/app/infobox/infobox.component.ts [hunk 5]
```diff
@@ -24,11 +25,9 @@ export class InfoboxComponent implements OnInit {
     this.content_response$ = store.select(fromRoot.getKnowledgeContent);
     this.content_response$.subscribe(res => {
       if (res.extract) {
-      this.title = res.title;
-      this.description = res.extract;
-      } else {
-        this.title = '';
-        this.description = '';
+        this.results = res;
```
Comment (simsausaurabh) [inline src/app/infobox/infobox.component.ts]: should be ```results$```.
